### PR TITLE
Fix in-progress check for utils/deepCircularCopy

### PR DESCRIPTION
**Pull request by afterwind-io**
## Changes

Lately we are working on IE11 support and found this weird problem.

TL;DR
After applying babel polyfill for `Symbol`, if you try to create a `Symbol`, whatever it creates is written to the prototype chain, which leads to...

```javascript
var a = Symbol("__I_am_a_symbol__");
var b = {};

console.log(a in b); // true
console.log(a[b]); // undefined
```

As a result, it brakes the in-progress check in `deepCircularCopy`:
https://github.com/PostHog/posthog-js/blob/d939de99893b1f2c2d9287d7d516bdb021908430/src/utils.js#L306

...and causes error in
https://github.com/PostHog/posthog-js/blob/d939de99893b1f2c2d9287d7d516bdb021908430/src/posthog-core.js#L577

...because `data` is `undefined`.

This PR takes the edge case into consideration, and basically does not change the origin logic.

## Checklist
- [x] Tests for new code (if applicable)
- [ ] TypeScript definitions (module.d.ts) updated and in sync with library exports (if applicable)

### CODE REVIEW

**Comment by macobo** [comment]
Would you mind adding an appropriate test to `src/__tests__/utils.js`? this can be under tests for copyAndTruncateStrings or for this function specifically

**Comment by afterwind-io** [comment]
Working on it.

**Comment by afterwind-io** [comment]
Test added. 

For convenience, `COPY_IN_PROGRESS_ATTRIBUTE` is exported from the `utils.js`, and I am not sure if this may violate your code guidelines, so I added some extra comments.

File: src/__tests__/utils.js [hunk 1]
```diff
@@ -62,6 +62,16 @@ describe('_.copyAndTruncateStrings', () => {
...
     })
+
+    it('should check copy-in-progress correctly', () => {
+        given('target', () => {
+            const base = Object.create({ [COPY_IN_PROGRESS_ATTRIBUTE]: undefined })
+            const object = Object.create(base)
+            return object
+        })
+
+        expect(given.target).not.toEqual(void 0)
```
Comment (macobo) [inline src/__tests__/utils.js:73]: This test does not seem to test neither deepCircularCopy or copyAndTruncateStrings in it's current form.

Did you intend something like:
```
expect(given.subject).toEqual({})
```
?

Comment (afterwind-io) [reply src/__tests__/utils.js:73]: Oops, my bad. 🤣

**Comment by afterwind-io** [comment]
Done.

**Comment by macobo** [comment]
Thanks! :100:
